### PR TITLE
[Feature] Adding Pre-commit hook and Prettier

### PR DESCRIPTION
**Pull request by SBoudrias**
The prettier integration is done with eslint as I found it to be the more convenient. It allows us to easily enforce the code style in the CI process.

Rel #259

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,11 +51,32 @@
     "yeoman-generator": "^1.1.1"
   },
   "eslintConfig": {
-    "extends": "xo-space",
+    "extends": [
+      "xo-space",
```
Comment (luftywiranda13) [inline package.json]: xo-space isn't needed here.. already interfered by `eslint-config-prettier` and `eslint-plugin-prettier`

Comment (SBoudrias) [reply package.json]: `eslint-config-prettier` only turn off style related rules.

`xo-space` add other linting rules. That being said, we probably could just use the `xo` preconfigured rules.

Comment (luftywiranda13) [reply package.json]: Yeah, but we're also using `eslint-plugin-prettier` which prefers space by default. It makes `xo-space` useless to be used here, doesn't it?

Comment (SBoudrias) [reply package.json]: `xo-space` is the `xo` config + space preference. So yeah, we don't specifically need xo-space, we can use `xo` directly.

Comment (luftywiranda13) [reply package.json]: ah my bad, I misread this..
you're using `xo` as a `eslint` shareable config, not directly 😞

Comment (luftywiranda13) [reply package.json]: but yeah, why use eslint-config-xo-space instead of eslint-config-xo? directly?

**Pull request merged**